services/QuillLMS/app/services/demo/session_data.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Demo
+  class SessionData
+    DEMO_PATH = 'lib/data/demo'
+    FILE_DIRECTORY = Rails.root + DEMO_PATH
+
+    # The parsed yaml files need access to the Models not nested in modules to load properly
+    ActivitySession = ::ActivitySession
+    ConceptResult = ::ConceptResult
+    ConceptResultQuestionType = ::ConceptResultQuestionType
+
+    FILE_ACTIVITY_SESSION = 'activity_sessions.yml'
+    FILE_CONCEPT_RESULTS = 'concept_results.yml'
+    FILE_CONCEPT_RESULT_QUESTION_TYPES = 'concept_result_question_types.yml'
+    FILE_CONCEPT_RESULT_LEGACY_METADATA = 'concept_result_legacy_metadata.yml'
+
+    ACTIVITY_USER_PAIRS = Demo::ReportDemoCreator::ACTIVITY_PACKS_TEMPLATES
+      .map {|hash| hash[:activity_sessions].map(&:to_a)}
+      .flatten(2)
+      .push([Demo::ReportDemoCreator::REPLAYED_ACTIVITY_ID, Demo::ReportDemoCreator::REPLAYED_SAMPLE_USER_ID])
+
+    def activity_sessions
+      @activity_sessions ||= load_file(FILE_ACTIVITY_SESSION)
+    end
+
+    def concept_results
+      @concept_results ||= load_file(FILE_CONCEPT_RESULTS)
+    end
+
+    def concept_result_question_types
+      @concept_result_question_types ||= load_file(FILE_CONCEPT_RESULT_QUESTION_TYPES)
+    end
+
+    def concept_result_legacy_metadata
+      @concept_result_legacy_metadata ||= load_file(FILE_CONCEPT_RESULT_LEGACY_METADATA)
+    end
+
+    private def load_file(file)
+      YAML.load_file(FILE_DIRECTORY + file)
+    end
+
+    # Connect to Prod Follower and run
```

Comment:
I think this entry point should be more heavily documented.  Some possibilities:
- bootstrap.sh
- a new rake file

Also, if the record ids are sufficiently parameterized, we don't need to run this specifically on prod. Any up to date database (e.g. staging) would be sufficient.

Reply:
I've updated the comments, since I don't think this was clear. This `generate_files` should rarely, if ever be run. It will only be run if someone wants to change the data that is in the demo account (we've done this twice in the past 3+ years). This PR takes the info from the sessions and stores it in the repo. We then use those files to generate the demo data, to allow us to have consistent demo data in local, staging, and prod environments. It's also a lot faster!